### PR TITLE
Add Stably GH Workflow

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

This PR Introduces an update to our `preview` GH action that adds a job to run our Stably AI test suite after the `preview` job runs successfully. The new job is dependent on the `preview` job as we're using that environment in order to run the tests. This way, when the preview environment is cleaned up, so are the artifacts created by the tests.

### Testing

- Run the test suite and run a test that knowingly fails, and ensure it blocks CI.
- Run the test suite and run a test that will pass, and ensure that it passes correctly
- Run the test suite and ensure that if the `preview` job fails, we don't attempt the `Stably Test Runner` job.

### CODE REVIEW

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -16,16 +16,49 @@ jobs:
...
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Stably Runner Action
+        id: stably-runner
+        uses: stablyhq/stably-runner-action@v3
+        with:
+          api-key: ${{ secrets.STABLY_API_KEY }}
+          run-group-ids: |-
+            clp2xrhh60001l608s1b4w4zd
```
Comment (mknowlton89) [inline .github/workflows/preview.yml:49]: I created a new "Test Group" so we can control exactly what tests run via this GH action. Currently, there is only a single test, but we can update this suite without needing to update the GH action.

Comment (mknowlton89) [reply .github/workflows/preview.yml:49]: Currently, we're running into a timeout issue with a lot of our main tests - Stably is addressing that today.

**Comment by bttf** [comment]
Github workflow config looks good 👍🏾

This will run for every pull request we push (just as preview environments do). Just want to make sure that this won't run into any constraints from Stably's side...

Seems like it would be ideal to add a shortlist of basic tests to start us off with. Such as logging in via email, creating an experiment, creating a feature, etc... 

One issue I'm seeing with some of the current tests is that they aren't making any assertions - for example, the login with invalid email test is passing but it's not asserting that there is specific UI state after clicking sign up. The same is occurring for the login with email test.

**Comment by mknowlton89** [comment]
> Github workflow config looks good 👍🏾
> 
> This will run for every pull request we push (just as preview environments do). Just want to make sure that this won't run into any constraints from Stably's side...
> 
> Seems like it would be ideal to add a shortlist of basic tests to start us off with. Such as logging in via email, creating an experiment, creating a feature, etc...
> 
> One issue I'm seeing with some of the current tests is that they aren't making any assertions - for example, the login with invalid email test is passing but it's not asserting that there is specific UI state after clicking sign up. The same is occurring for the login with email test.

Awesome - I've just followed up with the Stably team just to double check re: constraints on their side.

And yes, I agree that we should polish an initial set of tests to run before launch. I can take some time today to do so.

**Comment by mknowlton89** [comment]
Closing due to age.

**Pull request closed without merging**